### PR TITLE
Added Quick Sort in Java

### LINKED ISSUE #451: Quicksort in Java
#### This issue is about [Algo/DS Name](link to resource for the Algo/DS)

-   [x] Issue Title is in form `[Algo/DS Name] in [Language]` (ex:- BubbleSort in Java)

-   [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate

-   [x] Assign this issue to me (I want to work on this)

   -   [x] I ensure that I am not already assigned to 2 or more issues
  
-   [x] I will send my PR only after I'm assigned to this issue by any maintainer

### PR DESCRIPTION
**Pull request by mertbarutcuoglu**
**Fixes #451 **

By submitting this pull request I confirm I've read and complied with the below declarations.

-   [x] I have read the [Contribution guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md) 
and I am confident that my PR reflects them.

-   [x] I have followed the [coding guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#cs) 
for this project.

-   [x] My code follows the [skeleton code structure](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#sample).

-   [x] This pull request has a descriptive title. 
For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

-   [x] This pull request shall only be merged once the build passes.

-   [x] I have mentioned the issue number correctly (with hyper-link) in this pull request description.

### CODE REVIEW

File: algorithms/ar-qsrt/java/QuickSort.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+        int mid = (left + right) / 2;
+        int pivot = arr[mid];
+
+        int partitionIndex = partition(left, right, pivot, arr);
+        quicksort(left, partitionIndex - 1, arr);
+        quicksort(partitionIndex, right, arr);
+    }
+
+    private static int partition(int left, int right, int pivot, int[] arr) {
```
Comment (RavinduSachintha) [inline algorithms/ar-qsrt/java/QuickSort.java:20]: Codacy found an issue: [Avoid reassigning parameters such as 'left'](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6321063)

Comment (RavinduSachintha) [inline algorithms/ar-qsrt/java/QuickSort.java:20]: Codacy found an issue: [Avoid reassigning parameters such as 'left'](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6321063)

Comment (mertbarutcuoglu) [reply algorithms/ar-qsrt/java/QuickSort.java:20]: @RavinduSachintha Should I create temporary variables to resolve the Codacy issue or is it okay?

Comment (Danushka96) [reply algorithms/ar-qsrt/java/QuickSort.java:20]: @mertbarutcuoglu Use temporary local variables.

Comment (mertbarutcuoglu) [reply algorithms/ar-qsrt/java/QuickSort.java:19]: @Danushka96 Done.

**Review by RavinduSachintha — APPROVED** [review]

**Comment by RavinduSachintha** [comment]
@mertbarutcuoglu good job :+1:

**Pull request merged**